### ISSUE TITLE
Bug on f6dcc7c with something

### ISSUE THREAD

**Issue by Deophius** [Sudoku-dev/Sudoku#55]
On commit _f6dcc7c_

## Description
When I was testing, it just exited.
Using `gdb`, I saw that it was because of SIGSEGV.

**Comment by Deophius** [comment]
Let's first make sure that the testing file doesn't cause the error.

**Comment by Deophius** [comment]
Well, it does.
I forgot to associate the fieldset to the board.
Let's fix it and try again.
I don't think this will be the only bug.

**Comment by Deophius** [comment]
Still wrong.
With a little tracing, I found that `find_one_solution` actually completed in less than 100ms.
The problem is in `print_board_with_result`.

**Comment by Deophius** [comment]
I forgot to add `.value()` in `print_board_with_result`, so it couldn't output the cell correctly.
But after I added it, the output was so ridiculous:
input: 
7 0 3 0 0 0 0 0 0
0 0 6 0 0 1 7 8 0
0 4 0 0 0 2 0 6 0
6 0 0 0 0 0 1 7 0
9 0 0 1 6 0 0 0 0
3 0 2 0 9 0 0 0 0
0 0 0 4 0 0 0 1 8
0 7 0 0 0 0 4 0 0
0 0 0 6 8 5 0 0 7
output:
1 1 1 5 4 4 2 2 1
2 2 5 3 3 1 2 2 2
1 1 1 3 3 2 3 3 1
4 5 4 2 2 3 1 2 2
4 5 4 1 2 3 2 2 2
1 1 1 5 4 4 5 4 4
2 2 5 2 2 3 2 1 2
1 1 1 2 1 3 2 2 2
1 1 1 2 1 3 2 2 2

**Comment by Deophius** [comment]
It seemed that now the `print` is correct.
But why does it still produce messy output?
Perhaps that is because the solve functions are wrong.

**Comment by Deophius** [comment]
When I stuck a whole solved Sudoku puzzle into the program:
7 2 3 8 5 6 9 4 1
5 9 6 3 4 1 7 8 2
1 4 8 9 7 2 3 6 5
6 8 4 5 2 3 1 7 9
9 5 7 1 6 4 8 2 3
3 1 2 7 9 8 6 5 4
2 6 9 4 3 7 5 1 8
8 7 5 2 1 9 4 3 6
4 3 1 6 8 5 2 9 7
count_solutions still refuses to work.

**Comment by Deophius** [comment]
It seemed that problem was only once-occuring.
Now it's normal again.

**Comment by Deophius** [comment]
That must be a security vulneralbility. I'll discuss it in another issue

**Comment by Deophius** [comment]
Let's try to add tracing statements in the middle of the solve functions.
That should work, I hope.

**Comment by Deophius** [comment]
I get it! It's now working well with the first input.
See commit 8b11c0a for details.

**Comment by Deophius** [comment]
Now `find_one_solution` has passed 3 tests. Perhaps we can put it into "awaiting merge"?
